Get also none product image thumbnails via API

cewald opened the pull request:
### Related issues
closes #3207

### Short description and why it's useful

Adds the possibility to get also none product image thumbnails via API using a special parameter in `getThumbnail()` or using `getMediaThumbnail()`.

It's enabled by default, which I'm not sure about because of the needed change of the API image-url. But it is backwards compatible if you disable it in configuration.

I also added some types to the `getThumbnailPath()` method parameters in `core/helpers/index.ts`.

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance
<!-- Please get familiar with following info -->

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

patzick commented on core/helpers/index.ts:
```diff
@@ -28,26 +28,34 @@ export function slugify (text) {
   if (config.images.useExactUrlsNoProxy) {
     return relativeUrl // this is exact url mode
   } else {
+    if (config.images.hasOwnProperty('useSpecificImagePaths') && config.images.useSpecificImagePaths) {
```
```suggestion
    if (config.images.useSpecificImagePaths) {
```

patzick commented on core/mixins/thumbnail.js:
```diff
@@ -3,13 +3,20 @@ import { getThumbnailPath as _thumbnailHelper } from '@vue-storefront/core/helpe
-    getThumbnail (relativeUrl, width, height) {
-      return _thumbnailHelper(relativeUrl, width, height)
-    }
+    getThumbnail: (relativeUrl, width, height, pathType) => _thumbnailHelper(relativeUrl, width, height, pathType),
```
we should change `as` in import and use original helper name
```suggestion
    getThumbnail: (relativeUrl, width, height, pathType) => getThumbnailPath(relativeUrl, width, height, pathType),
```

patzick commented on core/mixins/thumbnail.js:
```diff
@@ -3,13 +3,20 @@ import { getThumbnailPath as _thumbnailHelper } from '@vue-storefront/core/helpe
+     * @param {number} width
+     * @param {number} height
+     */
+    getMediaThumbnail: (relativeUrl, width, height) => _thumbnailHelper(relativeUrl, width, height, 'media')
```
```suggestion
    getMediaThumbnail: (relativeUrl, width, height) => getThumbnailPath(relativeUrl, width, height, 'media')
```

cewald commented:
I've updated the code with your suggestions and removed the empty "media" config and forced it to be the default code if the `pathType` is not found in the map.

pkarw commented:
Seemingly we do have a duplicate on this feature with: https://github.com/DivanteLtd/vue-storefront/pull/3224

Can you please discuss with Resubaka the optimal way to merge these both changes in? You’ve figured out how to maintain the backward compatibility - I like it - would be great to use the best of both your PRs into one :)

ResuBaka commented on core/helpers/index.ts:
```diff
@@ -28,26 +28,32 @@ export function slugify (text) {
     return relativeUrl // this is exact url mode
   } else {
+    if (config.images.useSpecificImagePaths) {
+      const path =  (!Object.keys(config.images.paths).includes(pathType)) ? '' : config.images.paths[pathType]
```
```typescript
const path = config.images.paths[pathType] !== undefined ? config.images.paths[pathType] : ''
```

For me it is more readable and easier to understand when i write it like this.

ResuBaka commented on config/default.json:
```diff
@@ -372,6 +372,10 @@
     "images": {
       "useExactUrlsNoProxy": false,
       "baseUrl": "https://demo.vuestorefront.io/img/",
+      "useSpecificImagePaths": true,
```
Here it should use false as default value and the documentation should then say we need to change the base image URL in the API to have that working.

cewald replied:
Yes definitely, this was my fault.

cewald commented:
@ResuBaka I've added your changes.

patzick commented:
@ResuBaka could you re-review?

ResuBaka commented on core/mixins/thumbnail.js:
```diff
@@ -1,15 +1,22 @@
+     * @param {number} width
+     * @param {number} height
+     */
+    getMediaThumbnail: (relativeUrl, width, height) => getThumbnailPath(relativeUrl, width, height, 'media')
```
Shouldn't we have some kind of default config for it other wise i don't know how this can ever function. 

So like this: `"media": "/"`?

cewald replied:
In the docs we tell that the you have to remove `/catalog/product` from the API's `magento1.imgUrl` or `magento2.imgUrl` config property so whats then left will be `http://magento-demo.local/media` as default value and "root-path". That's why we don't need `"media": "/"` anymore.

ResuBaka replied:
Yea your are right I was probably only a bit confused that we use the media pathType which is not set in the config. :smile:
